pkg/sql/logictest/testdata/logic_test/jobs

```diff
@@ -188,3 +188,16 @@ usage_count > 0
 ORDER BY feature_name DESC
 ----
 job.schema_change.successful
+
+# Validate that the virtual index on 'status' works.
+# Vectorized execution may be on or off during tests, so exclude it from the output.
+query T
+SELECT info FROM [EXPLAIN SELECT count(*) FROM crdb_internal.jobs WHERE status = 'paused'] WHERE info NOT LIKE 'vectorized%'
```

Comment:
gonna review a bit more soon, but i think we'd want this test in pkg/sql/opt/exec/execbuilder/testdata/virtual

https://github.com/cockroachdb/cockroach/pull/90597/files#diff-34643f34113d939ff0d8b4ef11a6d54d286a0a413851aebbc544013d2c071f88